rust/hyperlane-core/src/accumulator/historic.rs

```diff
@@ -0,0 +1,155 @@
+use crate::{
+    accumulator::{hash_concat, merkle::Proof, ZERO_HASHES},
+    H256,
+};
+
+use super::{merkle::MerkleTree, TREE_DEPTH};
+
+impl MerkleTree {
```

Comment:
wdyt of a `generate_historic_proof` or something rather than having the prover do it?

Reply:
No real preference either way, will do